src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -144,6 +145,38 @@ public int numOfSessionBetween(SessionDate inclusiveStart, SessionDate inclusive
         return (firstInSpan.numOfDayTo(lastInSpan) / interval.getValue()) + 1;
     }
 
+    /**
+     * Checks if the {@code Session} slot overlaps with all recurring sessions.
+     * @param otherSession the other session that is compared to.
+     */
+    public boolean isOverlapping(Session otherSession) {
+        return otherSession.isOverlapping(this);
+    }
+
+    /**
+     * Checks if all {@code RecurringSession} slots overlaps with all recurring sessions.
+     * @param otherSession the other recurring session that is compared to.
+     */
+    public boolean isOverlapping(RecurringSession otherSession) {
+        if (startAfter(otherSession.getSessionDate())) {
+            return otherSession.isOverlapping(this);
+        }
+        int daysBetween = (int) ChronoUnit.DAYS
+                .between(getSessionDate().getDate(), otherSession.getSessionDate().getDate());
+
+        int intervalMod = getInterval().getValue() % otherSession.getInterval().getValue();
+        int startDiffMod = daysBetween % otherSession.getInterval().getValue();
+        int diffMod = intervalMod % startDiffMod;
+        if (diffMod != 0) {
+            return false;
```

Comment:
eg.
daysBetween = 8,
interval1 = 3, interval2 = 7 vs. interval1 = 7, interval2 = 3
seems to give false n true respectively with this method.

by right both cases should be false

Reply:
i suggest
public boolean isOverlapping(RecurringSession otherSession) {
        if (startAfter(otherSession.getSessionDate())) {
            return otherSession.isOverlapping(this);
        }
        Session firstAfter = this.firstSessionOnOrAfter(otherSession);
        int daysBetween = otherSession.getSessionDate().daysTo(firstAfter.getSessionDate());
        if (daysBetween % gcd(this.getInterval, otherSession.getInterval)  != 0) {
             return false;
        }

Reply:
SORRY i take the daysBetween complication back.
a simple daysBetween using existing `lastSessionOnOrBefore` will do.

i suggest
public boolean isOverlapping(RecurringSession otherSession) {
if (startAfter(otherSession.getSessionDate())) {
return otherSession.isOverlapping(this);
}
Session lastBefore = this.lastSessionOnOrBefore(otherSession);
int daysBetween = lastBefore.daysTo(otherSession.getSessionDate());
if (daysBetween % gcd(this.getInterval, otherSession.getInterval) != 0) {
return false;
}
......
ur time thing

Reply:
gcd(interval 1, interval2) is a factor of daysBetween  <- -> will overlap in future
because
And overlap existing at some point means,
daysBtw + (i2 x n) = i1 x m, for real number n and m.

diophantine equation ax+by=c has solutions if and only if gcd(a,b)|c ((meaning c is divisible by gcd(a,b)))
c being our daysBtw, i1, i2 being a and b respectively